pkg/build/generator/generator.go

```diff
@@ -702,3 +710,18 @@ func getStrategyImageChangeTrigger(bc *buildapi.BuildConfig) *buildapi.ImageChan
 	}
 	return nil
 }
+
+// getImageChangeTriggerForRef returns the ImageChangeTrigger that is triggered by a change to
+// the provided object reference, if any
+func getImageChangeTriggerForRef(bc *buildapi.BuildConfig, ref *kapi.ObjectReference) *buildapi.ImageChangeTrigger {
+	if ref == nil || ref.Kind != "ImageStreamTag" {
+		return nil
+	}
+	for _, trigger := range bc.Spec.Triggers {
+		if trigger.Type == buildapi.ImageChangeBuildTriggerType && trigger.ImageChange.From != nil &&
```

Comment:
yes... but resolveImageStreamReference would get called on the source image from, instead of using the id from the trigger, which is what I thought you were trying to do with this PR.